### ISSUE TITLE
most logs are outputting through `stderr` instead of `stdout` 

### ISSUE THREAD

**Issue by Connoropolous** [holochain/holochain#1449]
When I run the following:
`RUST_LOG=debug holochain 2> capture.log`

Here is what appears in the console directly:
```
###HOLOCHAIN_SETUP###
###ADMIN_PORT:1234###
###HOLOCHAIN_SETUP_END###
Conductor ready.
```
These appear because they are output via `println`. 

Here is what appears in the file `capture.log`, which is a capture of the `stderr` stream.
```
[2mJun 20 12:37:29.112[0m [34mDEBUG[0m holochain: observability initialized
[2mJun 20 12:37:29.112[0m [34mDEBUG[0m holochain: config_path: /Users/connor/Library/Preferences/org.holochain.holochain/conductor-config.yml
[2mJun 20 12:37:29.112[0m [32m INFO[0m holochain::conductor::conductor::builder: self.config=ConductorConfig { environment_path: DatabaseRootPath("/Users/connor/Library/Application Support/org.holochain.holochain/databases"), keystore: LairServerLegacyDeprecated { keystore_path: None, danger_passphrase_insecure_from_config: "default-insecure-passphrase" }, dpki: None, admin_interfaces: Some([AdminInterfaceConfig { driver: Websocket { port: 1234 } }]), network: None, db_sync_strategy: Fast }
[2mJun 20 12:37:29.113[0m [33m WARN[0m holochain::conductor::conductor::builder: Using DEPRECATED legacy lair api.
[2mJun 20 12:37:29.113[0m [33m WARN[0m holochain::conductor::conductor::builder: USING INSECURE PASSPHRASE FROM CONFIG--This defeats the whole purpose of having a passphrase.
[2mJun 20 12:37:29.113[0m [31mERROR[0m lair_keystore_client: error=IpcClientConnectError("/Users/connor/Library/Application Support/host.Holo.Lair/socket", Os { code: 61, kind: ConnectionRefused, message: "Connection refused" }) file="/private/tmp/nix-build-hc_holochain_kitsune-p2p-tx2-proxy.drv-0/cargo-vendor-dir/lair_keystore_client-0.0.10/src/lib.rs" line=41
[2mJun 20 12:37:29.592[0m [32m INFO[0m holochain_sqlite::schema: database initialized: Conductor
[2mJun 20 12:37:29.594[0m [32m INFO[0m holochain_sqlite::schema: database initialized: Wasm
[2mJun 20 12:37:29.594[0m [32m INFO[0m kitsune_p2p_types::tx2::mem: bound local endpoint (mem) url=kitsune-mem://1
[2mJun 20 12:37:29.594[0m [32m INFO[0m kitsune_p2p_types::tx2::tx2_restart_adapter: tx2_restart_adapter bound Ok(kitsune-mem://1)
[2mJun 20 12:37:29.594[0m [34mDEBUG[0m [1mbind_with_handle[0m: holochain_websocket::websocket_listener: sever_listening_on=Url2 { url: "ws://127.0.0.1:1234/" }
[2mJun 20 12:37:29.594[0m [32m INFO[0m holochain::conductor::manager: Task added. Total tasks: 2
[2mJun 20 12:37:29.595[0m [34mDEBUG[0m holochain::conductor::handle: msg="Processing app status delta" delta=SpinUp
[2mJun 20 12:37:29.595[0m [34mDEBUG[0m holochain::conductor::handle: msg="Processing app status delta" delta=NoChange
[2mJun 20 12:37:29.595[0m [32m INFO[0m holochain: Conductor successfully initialized.
```

We can see the use of `info!` for example is outputting through `stderr`. 
https://github.com/holochain/holochain/blob/c5433dfc5f20f82f95d0631cd7ffaa6f74708515/crates/holochain/src/bin/holochain/main.rs#L91

**Comment by Connoropolous** [comment]
Tagging @jost-s because they are dealing with related comments that I've made downstream in `tryorama` which is equally affected by this, as will be many other applications. 
https://github.com/holochain/tryorama/issues/124

**Comment by Connoropolous** [comment]
The desired behaviour, I imagine, is that only `error!` are put through `stderr` but I am not quite sure. There may be other implications around that I don't know about.

**Comment by abe-njama** [comment]
ping @jost-s , you'd been tagged on this issue initially. Mind forming an opinion on the reported issue?

**Comment by maackle** [comment]
I have seen articles making a case that stdout should be for "normal program output", and stderr should be for any diagnostic message, not only errors, including logging and tracing. I tend to agree with this, since some commands like `holochain -i` really does rely on stdout to interact with the user, and logs should not clutter up that stream.

But, it seems there is no right way or consensus on the matter on the Internet, so we can decide whatever we want. I don't feel strongly about it either way.

**Comment by maackle** [comment]
@Connoropolous @jost-s could you summarize the reasons why it's desired for tracing logs to go to stdout instead of stderr? What is the desired experience, and for whom? I'm wondering if that outcome can be achieved in another way. For instance, if it's desired that every end user sees warnings and logs in their stdout, perhaps we can duplicate those logs to stdout and stderr.

I'm wondering because there are benefits to keeping stdout clean. I recently wrote an extensive TUI for diagnostics which uses stdout, which we could some day include in the conductor binary directly. Imagine running `holochain --diagnostics` and seeing a `htop` like display of all conductor activity in the terminal. This isn't a definite plan but it's something that becomes a bit harder once stdout is expected to contain traces. Tracing can always be redirected to different streams in different modes, but I guess it's about expectations and conventions.

**Comment by Connoropolous** [comment]
Due to a shift in my personal focus away from development towards business development, I no longer have commentary or thoughts on the issue. Going to unsub from this one :)

**Comment by ThetaSinner** [comment]
I'm inclined to agree with @maackle here. In the straight forward case of launching `holochain` in a terminal you'll see everything anyway. For things that have a reason to split the stdout/stderr streams there is a clear distinction. Actually really everything except for the admin port and the ready message probably belong on stderr in that case.

@jost-s do you have an opinion on this?

**Comment by jost-s** [comment]
We're all in agreement then, and secondly the root cause for this issue has been resolved as well. Tryorama outputs all tracing logs printed from a zome function to stdout, regardless of log level and not requiring any env vars like RUST_LOG to be set.
